ergotree-ir/src/mir/header.rs

```diff
@@ -20,6 +22,47 @@ pub struct PreHeader {
     pub votes: Vec<u8>,
 }
 
+/// Represents data of the block header available in Sigma propositions.
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct Header {
+    /// Block version, to be increased on every soft and hardfork.
+    pub version: u8,
+    /// Bytes representation of ModifierId of this Header
+    pub id: Vec<u8>,
+    /// Bytes representation of ModifierId of the parent block
+    pub parent_id: Vec<u8>,
+    /// Hash of ADProofs for transactions in a block
+    pub ad_proofs_root: Vec<u8>,
```

Comment:
`Digest32` should be enough for now, we'll add a specific domain type later when(if) we need to deal with AD proofs here. Check out ergo https://github.com/ergoplatform/ergo/blob/e68ce6180b13bffb024cf9e26c7c16a7be70a22c/src/main/scala/org/ergoplatform/modifiers/history/Header.scala#L57-L87 for more precise types.